libazureinit/src/provision/ssh.rs

```diff
@@ -2,17 +2,26 @@
 // Licensed under the MIT License.
 
 use std::{
-    fs::Permissions,
-    io::{self, Write},
+    fs::{self, OpenOptions, Permissions},
+    io::{self, Read, Write},
     os::unix::fs::{DirBuilderExt, PermissionsExt},
     path::PathBuf,
     process::Command,
 };
 
+use regex::Regex;
+use tempfile::NamedTempFile;
 use tracing::{info, instrument};
 
 use crate::error::Error;
 use crate::imds::PublicKeys;
+use lazy_static::lazy_static;
+
+lazy_static! {
+    static ref PASSWORD_REGEX: Regex =
+        Regex::new(r"(?m)^\s*#?\s*PasswordAuthentication\s+(yes|no)\s*$")
+            .unwrap();
```

Comment:
```suggestion
            .expect("The regular expression is invalid or exceeds the default regex size");
```
as @jeremycline suggested.